lxd/instances_post.go

```diff
@@ -863,12 +862,7 @@ func instancesPost(d *Daemon, r *http.Request) response.Response {
 			return api.StatusErrorf(http.StatusBadRequest, "Target only allowed when clustered")
 		}
 
-		var targetMember, targetGroup string
-		if strings.HasPrefix(target, "@") {
-			targetGroup = strings.TrimPrefix(target, "@")
-		} else {
-			targetMember = target
-		}
+		_, targetGroup := shared.TargetDetect(queryParam(r, "target"))
```

Comment:
Is `targetGroup` defined too early now, its only used with `project.CheckTarget` later right?

I'm wondering if it would be clearer for project.CheckTarget to return 2 values, the node info if the target was a specific node, and the group name if the target was a group.

Then you could do:

```
targetMemberInfo, targetGroupName, err := project.CheckTarget(ctx, r, tx, targetProject, queryParam(r, "target"), allMembers)
```

This seems clearer as it keeps all of that logic together. As having ` project.CheckTarget` return `nil,nil` when the target is a group seems a bit unusual.